### PR TITLE
story(ccls-1975) proceedings and costs summary

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
Includes changes for:

- ccls-1975 - proceedings and costs
- ccls-1957 - amend client details
- ccls-2034 - amend nullifying strings
- Updated GOV UK and MOJ design system templates

### CODE REVIEW

File: src/main/java/uk/gov/laa/ccms/caab/client/CaabApiClient.java [hunk 1]
```diff
@@ -251,4 +330,29 @@ public Mono<Void> putApplication(
...
         .onErrorResume(e -> caabApiClientErrorHandler.handleUpdateApplicationError(e, type));
   }
+
+  /**
+   * Patches all application clients with the same reference id using the CAAB API.
+   *
+   * @param clientReferenceId the ID associated with the client
+   * @param loginId the ID associated with the user login
+   * @param data the client data to amend to the application
+   * @return a Mono containing application's data
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/client/CaabApiClient.java:340]: It returns a Void

Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/client/CaabApiClient.java]: Amended

File: src/main/resources/templates/application/summary/proceedings-and-costs-section.html [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                <img class="govuk-!-padding-left-2 govuk-!-padding-right-2" src="/tick.gif" alt="tick">
+              </td>
+              <td class="govuk-table__cell"></td>
+            </th:block>
+
+            <th:block th:unless="${proceeding.leadProceedingInd}">
+              <td class="govuk-table__cell"></td>
+              <td class="govuk-table__cell">
+                <a th:href="@{/application/summary/proceedings/{id}/remove(id=${proceeding.id})}"
+                   class="govuk-link" th:text="${'Delete'}">
```
Comment (porritta) [inline src/main/resources/templates/application/summary/proceedings-and-costs-section.html:77]: Any reason for these being thymeleaf text when they're static strings?

File: src/main/resources/templates/application/summary/proceedings-and-costs-section.html [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+           href="/todo">
+          Add proceeding
+        </a>
+      </div>
+      <div class="govuk-tabs__panel govuk-tabs__panel--hidden" id="case-costs" aria-labelledby="tab_case-costs">
+        <h2 class="govuk-heading-l">Case costs</h2>
+        <table class="govuk-table">
+          <thead class="govuk-table__head">
+          <tr class="govuk-table__row">
+            <th scope="col" class="govuk-table__header">Case cots</th>
```
Comment (porritta) [inline src/main/resources/templates/application/summary/proceedings-and-costs-section.html:102]: I know some cases deal with children. But I think you meant 'costs' :-)

**Review by porritta — COMMENTED** [review]
Just those minor changes remaining

File: src/main/resources/templates/application/summary/proceedings-and-costs-section.html (hunk 2 above)
Comment (PhilDigitalJustice) [reply src/main/resources/templates/application/summary/proceedings-and-costs-section.html:77]: amended

File: src/main/resources/templates/application/summary/proceedings-and-costs-section.html (hunk 3 above)
Comment (PhilDigitalJustice) [reply src/main/resources/templates/application/summary/proceedings-and-costs-section.html:102]: amended

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**